### PR TITLE
FIX-#5580: BUG: 'AVG|SUM' is only valid on integer and floating point

### LINKED ISSUE #5580: BUG: HDK: 'AVG|SUM' is only valid on integer and floating point
### Modin version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the latest released version of Modin.

- [X] I have confirmed this bug exists on the main branch of Modin. (In order to do this you can follow [this guide](https://modin.readthedocs.io/en/stable/getting_started/installation.html#installing-from-the-github-master-branch).)

### Reproducible Example

```python
import modin.pandas as pd

df = pd.DataFrame({"A": [True, False]})
print(df.mean())
print(df.sum())
```

### Issue Description

The code above fails on HDK engine.

### Expected Behavior

```
A    0.5
dtype: float64
A    1
dtype: int64
```

### Error Logs

<details>

```python-traceback

modin/experimental/core/execution/native/implementations/hdk_on_native/hdk_worker.py", line 76, in _executeRelAlgJson
    res = rel_alg_executor.execute()
  File "_sql.pyx", line 103, in pyhdk._sql.RelAlgExecutor.execute
RuntimeError: AVG is only valid on integer and floating point

```

</details>

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : d3948666bf211c296f3a5640f23a9471cd4f54f0
python           : 3.10.8.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.15.79.1-microsoft-standard-WSL2
Version          : #1 SMP Wed Nov 23 01:01:46 UTC 2022
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

Modin dependencies
------------------
modin            : 0.18.0+52.gd3948666.dirty
hdk              : present

pandas dependencies
-------------------
pandas           : 1.5.2
numpy            : 1.24.1
pytz             : 2022.7.1
dateutil         : 2.8.2
setuptools       : 66.0.0
pip              : 22.3.1
Cython           : None
pytest           : 7.2.1
hypothesis       : None
sphinx           : 6.1.3
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : 
fastparquet      : None
fsspec           : 2022.11.0
gcsfs            : None
matplotlib       : 3.6.3
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : 3.0.10
pandas_gbq       : None
pyarrow          : 8.0.1
pyreadstat       : None
pyxlsb           : None
s3fs             : 2022.11.0
scipy            : 1.10.0
snappy           : None
sqlalchemy       : 1.4.46
tables           : None
tabulate         : None
xarray           : None
xlrd             : 2.0.1
xlwt             : None
zstandard        : None
tzdata           : None

</details>

### PR DESCRIPTION
**Pull request by AndreyPavlenko**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5580 
- [x] tests added and passing
- [ ] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

**Comment by ienkovich** [comment]
Let's merge #5563 and rebase this one.

File: modin/experimental/core/storage_formats/hdk/query_compiler.py [hunk 1]
```diff
@@ -793,3 +795,10 @@ def free(self):
     @property
     def dtypes(self):
         return self._modin_frame.dtypes
+
+
+def _check_int_or_float(op, dtypes):  # noqa: GL08
+    codes = "bBhHiIlLqQpPfd"
+    for t in dtypes:
+        if t.char not in codes:
```
Comment (dchigarev) [inline modin/experimental/core/storage_formats/hdk/query_compiler.py:803]:
```suggestion
    for t in dtypes:
        if not (pandas.api.types.is_integer_dtype(t) or pandas.api.types.is_float_dtype(t)).
```

**Review by dchigarev — COMMENTED** [review]
Don't we want to add the same int-float check for other aggregations?

File: modin/experimental/core/storage_formats/hdk/query_compiler.py (hunk 1 above)
Comment (AndreyPavlenko) [reply modin/experimental/core/storage_formats/hdk/query_compiler.py:803]: This code fails with RuntimeError: halffloat is not yet supported.

Comment (dchigarev) [reply modin/experimental/core/storage_formats/hdk/query_compiler.py:803]: I've run this code and it worked perfectly fine:
```python
>>> for t in np.typecodes["All"]:
...     print(t, np.dtype(t))
...     print("\t",pandas.api.types.is_float_dtype(np.dtype(t)))
...     print("\t",pandas.api.types.is_integer_dtype(np.dtype(t)))
...
```
What's the source of this `halffloat` you're using to get the error?

Comment (AndreyPavlenko) [reply modin/experimental/core/storage_formats/hdk/query_compiler.py:803]: This error is raised by HDK. Here is the reproducer - https://github.com/modin-project/modin/pull/5583/files#diff-924b045aa089c05b3a717bc9cea19dc8581e2b8351fcc3c0568a54a27ce8a146R2064

Comment (dchigarev) [reply modin/experimental/core/storage_formats/hdk/query_compiler.py:803]: Okay, but can we make this piece more human-readable though? Let's define a set of allowed numeric dtypes for HDK somewhere out of this function and then reference it here:
```python
# hdk_on_native/dataframe/utils.py
HDK_ALLOWED_NUMERIC = set(np.typecodes["AllInteger"] + np.typecodes["Float"]) - {np.dtype(np.float16).char}

def check_if_allowed_numeric(dtypes):
    ...
```

Comment (AndreyPavlenko) [reply modin/experimental/core/storage_formats/hdk/query_compiler.py]: Done.

File: modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py [hunk 2]
```diff
@@ -2061,6 +2061,19 @@ def test_uint_serialization(self):
...
+        exclude_codes = np.typecodes["Datetime"] + np.typecodes["Complex"] + "gSUVO"
+        supported_codes = set(all_codes) - set(exclude_codes)
+
+        def test(df, dtype_code, operation, **kwargs):
+            df = type(df)({"A": [0, 1], "B": [1, 0]}, dtype=np.dtype(dtype_code))
+            return getattr(df, operation)()
+
+        for c in supported_codes:
+            run_and_compare(test, data={}, dtype_code=c, operation="sum")
+            run_and_compare(test, data={}, dtype_code=c, operation="mean")
```
Comment (anmyachev) [inline modin/experimental/core/execution/native/implementations/hdk_on_native/test/test_dataframe.py:2075]:
```suggestion
            for op in ("sum", "mean"):
                run_and_compare(test, data={}, dtype_code=c, operation=op)

**Review by anmyachev — APPROVED** [review]

**Comment by anmyachev** [comment]
@AndreyPavlenko please reply to the [comment](https://github.com/modin-project/modin/pull/5583#pullrequestreview-1285551424).

**Comment by AndreyPavlenko** [comment]
> Don't we want to add the same int-float check for other aggregations?

I'll check if it's required for other operations.

**Review by dchigarev — APPROVED** [review]

**Pull request merged**